config/cache/v1alpha1/cache.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package gingersnap.config.cache.v1alpha1;
+import "apimachinery/pkg/api/resource/quantity.proto";
+
+// TODO  option java_package = define package naming for java
+// TODO  option java_outer_classname = define outer class name
+option java_multiple_files = true;
+// TODO option go_package = verify below value is ok
+option go_package = "api/v1alpha1";
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes the desired configuration for a Cache. Only DB Cache Service is supported atm
+message CacheSpec {
+  // Resource profile for the cache provider
+  ResourcesProfile resources = 1;
+  // DatasourceRef or a ServiceBindingRef (TODO clarify)
+  NamespacedRef dataSourceRef = 2;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a data source connection. A map is available for passing implementation specific
+// properties.
+message DataSourceSpec {
+  // Type and version of the underlaying DB. Needed to decide which drivers need to be used
+  DBType dbType = 1;
+  // Additional properties. DB specific
+  map<string,string> connectionProperties = 2;
+  // Reference to the ServiceBinding CR
+  NamespacedRef serviceBindingRef = 3;
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Describes a caching rule behaviours
+message EagerCachingRuleSpec {
```

Comment:
Maybe I misunderstand how the code generation works, so this might not make sense. I think we need this message and the `LazyCachingRuleSpec` in their own standalone files so that the generated *.go files are containing in their own file.

Reply:
The idea is to have the actual CR composed in the application (operator) just aggregating spec and status, this way Spec can be also used non-k8s application.  No need to have a 1 message 1 file design, though it's possible. atm I'm using 1 file so I can tweak the options at the beginning easily.